doc/examples/uuid.rst

```diff
@@ -0,0 +1,472 @@
+
+The :data:`~bson.binary.UuidRepresentation.STANDARD` representation
+enables cross-language compatibility by ensuring the same byte-ordering
+when encoding UUIDs from all drivers. UUIDs written by a driver with this
+representation configured will be handled correctly by every other provided
+it is also configured with the ``STANDARD`` representation.
+
+``STANDARD`` encodes native :class:`uuid.UUID` objects to
+:class:`~bson.binary.Binary` subtype 4 objects.
+
+.. _unspecified-representation-details:
+
+``UNSPECIFIED``
+^^^^^^^^^^^^^^^
+
+.. attention:: Starting in PyMongo 4.0,
+   :data:`~bson.binary.UuidRepresentation.UNSPECIFIED` will be the default
+   UUID representation used by PyMongo.
+
+The :data:`~bson.binary.UuidRepresentation.UNSPECIFIED` representation
+prevents the incorrect interpretation of UUID bytes by stopping short of
+automatically converting UUID fields in BSON to native UUID types. Loading
+a UUID When using this representation returns a :class:`~bson.binary.Binary`
+object instead. If required, users can coerce the decoded
+:class:`~bson.binary.Binary` objects into native UUIDs using the
+:meth:`~bson.binary.Binary.as_uuid` method and specifyin the appropriate
+representation format. The following example shows
+what this might look like for a UUID stored by the C# driver::
+
+  from bson.codec_options import CodecOptions, DEFAULT_CODEC_OPTIONS
+  from bson.binary import Binary, UuidRepresentation
+  from uuid import uuid4
+
+  # Using UuidRepresentation.CSHARP_LEGACY
+  csharp_opts = CodecOptions(uuid_representation=UuidRepresentation.CSHARP_LEGACY)
+
+  # Store a C#-formatted UUID
+  input_uuid = uuid4()
+  collection = client.testdb.get_collection('test', codec_options=csharp_opts)
+  collection.insert_one({'_id': 'foo', 'uuid': input_uuid})
+
+  # Using UuidRepresentation.UNSPECIFIED
+  unspec_opts = CodecOptions(uuid_representation=UuidRepresentation.UNSPECIFIED)
+  unspec_collection = client.testdb.get_collection('test', codec_options=unspec_opts)
+
+  # UUID fields are decoded as Binary when UuidRepresentation.UNSPECIFIED is configured
+  uuid_1 = uuid4()
+  document = unspec_collection.find_one({'_id': 'foo'})
+  decoded_field = document['uuid']
+  assert isinstance(decoded_field, Binary)
+
+  # Binary.as_uuid() can be used to coerce the decoded value to a native UUID
+  decoded_uuid = decoded_field.as_uuid(UuidRepresentation.CSHARP_LEGACY)
+  assert decoded_uuid == input_uuid
+
+Native :class:`uuid.UUID` objects cannot directly be encoded to
+:class:`~bson.binary.Binary` when the UUID representation is ``UNSPECIFIED``
+and attempting to do so will result in an exception::
+
+  unspec_collection.insert_one({'_id': 'bar', 'uuid': uuid4()})
+  Traceback (most recent call last):
+  ...
+  ValueError: cannot encode native uuid.UUID with UuidRepresentation.UNSPECIFIED. UUIDs can be manually converted to bson.Binary instances using bson.Binary.from_uuid() or a different UuidRepresentation can be configured.
```

Comment:
Should this error message link to this documentation page? I'm wondering if we can help point the user to the right place.

Reply:
Not a huge fan of hardcoding links into the codebase, but we can do it if you feel strongly about it.

Reply:
What about something like "see the documentation for UuidRepresentation for more information".

Reply:
Sounds good... Making this change.